src/library/scala/collection/IterableOnce.scala

```diff
@@ -1056,10 +1059,13 @@ trait IterableOnceOps[+A, +CC[_], +C] extends Any { this: IterableOnce[A] =>
     *
     *  @example  `List(1, 2, 3).mkString("(", "; ", ")") = "(1; 2; 3)"`
     */
-  def mkString(start: String, sep: String, end: String): String =
-    addString(new StringBuilder(), start, sep, end).toString
+  @inline final def mkString(start: String, sep: String, end: String): String =
+    if (isEmpty) start + end
+    else addString(new StringBuilder(), start, sep, end).result
 
   /** Displays all elements of this $coll in a string using a separator string.
+    *
+    * Delegates to addString, which can be overriden.
     *
     *  @param sep   the separator string.
     *  @return      a string representation of this $coll. In the resulting string
```

Comment:
Needs a comma after `In the resulting string`.

Reply:
I do not agree with these type of changes. I'm not changing that line.

Reply:
If you want your nice mkString to have ugly scaladoc, that's not a call I'm in a position to make. But I'm just making comments, not starting a review. Rest assured.